gym-service/src/test/java/org/example/integration/steps/trainee/TraineeSteps.java

```diff
@@ -0,0 +1,161 @@
+
+
+    private final TestContext testContext;
+    private final MockMvc mockMvc;
+    private final UserRepository userRepository;
+    private MvcResult mvcResult;
+
+    @Given("the trainee is not authenticated with username {string}")
+    public void theTraineeIsNotAuthenticated(String username) {
+        testContext.setJwtToken(null);
+    }
+
+    @When("the trainee requests the trainee profile")
+    public void theTraineeRequestsTheTraineeProfile() throws Exception {
+        mvcResult = mockMvc.perform(MockMvcRequestBuilders.get(TRAINEE_PROFILE_URL)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .header(HttpHeaders.AUTHORIZATION, AUTHORIZATION_HEADER + testContext.getJwtToken())
+                        .contentType(CONTENT_TYPE_JSON))
+                .andReturn();
+    }
+
+    @When("the trainee requests to change the password with a valid new password {string}")
+    public void theTraineeRequestsToChangeThePasswordWithAValidNewPassword(String password) throws Exception {
+        performPasswordChangeRequest(password);
+    }
+
+    @When("the trainee requests to change the password with an invalid new password {string}")
+    public void theTraineeRequestsToChangeThePasswordWithAnInvalidNewPassword(String password) throws Exception {
+        performPasswordChangeRequest(password);
+    }
+
+    private void performPasswordChangeRequest(String password) throws Exception {
+        String passwordJson = new JSONObject().put("password", password).toString();
+        mvcResult = mockMvc.perform(MockMvcRequestBuilders.put(CHANGE_PASSWORD_URL)
+                        .header(HttpHeaders.AUTHORIZATION, AUTHORIZATION_HEADER + testContext.getJwtToken())
+                        .content(passwordJson)
+                        .contentType(CONTENT_TYPE_JSON))
+                .andReturn();
+    }
+
+    @When("the trainee requests to update the profile with complete data:")
+    public void theTraineeRequestsToUpdateTheProfileWithCompleteData(DataTable dataTable) throws Exception {
+        performProfileUpdateRequest(dataTable);
+    }
+
+    @When("the trainee requests to update the profile with incomplete data:")
+    public void theTraineeRequestsToUpdateTheProfileWithIncompleteData(DataTable dataTable) throws Exception {
+        performProfileUpdateRequest(dataTable);
+    }
+
+    private void performProfileUpdateRequest(DataTable dataTable) throws Exception {
+        JSONObject requestBody = new JSONObject();
+        for (Map<String, String> columns : dataTable.asMaps(String.class, String.class)) {
+            requestBody.put("firstName", columns.get("firstName"));
+            requestBody.put("lastName", columns.get("lastName"));
+            requestBody.put("isActive", columns.get("isActive"));
+            requestBody.put("dateOfBirth", columns.get("dateOfBirth"));
+            requestBody.put("address", columns.get("address"));
```

Comment:
your data could be "incomplete". what happens when some columns are missing? should it be handled properly?